docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,186 @@
+calling ``optimizer.step()``.
+
+Gradient penalty
+""""""""""""""""
+
+A gradient penalty loss term also requires awareness of the scale factor.
+Gradient penalty demonstrates the following use cases:
+
+* Creating out-of-place gradients with :func:`torch.autograd.grad`
+* Correct interaction of gradient scaling with double-backward
+
+Here's how that looks for a simple L2 penalty::
+
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+
+        # We should scale outputs for the out-of-place backward pass
+        grad_params = torch.autograd.grad(scaler.scale(loss), model.parameters(), create_graph=True)
+
+        # In general, the penalty term may depend nonlinearly on the out-of-place gradients, so to be safe,
+        # manually unscale them before computing the penalty.  This unscale should be out-of-place and autograd-exposed.
+        grad_params = [p*(1./scaler.get_scale()) for p in grad_params]
+
+        # Compute the penalty term and add it to the loss
+        grad_norm = 0
+        for grad in grad_params:
+            grad_norm += grad.pow(2).sum()
+        grad_norm = grad_norm.sqrt()
+        loss = loss + grad_norm
+
+        # The usual scaling for backward will now accumulate leaf gradients that are appropriately scaled.
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+
+Gradient accumulation
+---------------------
+
+Gradient accumulation across iterations (between steps) can be implemented as follows::
+
+    scaler = AmpScaler()
+    ...
+    for i, (input, target) in enumerate(data):
+        output = model(input)
+        loss = loss_fn(output, target)
+        loss = loss/iters_to_accumulate
+        scaler.scale(loss).backward()
+        if (i + 1) % iters_to_accumulate == 0:
+            scaler.step(optimizer)
+            scaler.update()
+            optimizer.zero_grad()
+
+Switching gradient scaling on and off
+------------------------------------
+
+You can enable/disable gradient scaling globally (everywhere in the network a given :class:`AmpScaler` instance
+is used) by supplying a single ``enabled=True|False`` flag to that instance's constructor call::
+
+    scaler = AmpScaler(enabled=args.use_mixed_precision)
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+If ``enabled=False``, ``scaler.step(optimizer)`` is equivalent to ``optimizer.step()``, and
+the other methods (``scaler.scale``, ``scaler.update``) become no-ops.
+
+
+Working with Multiple Optimizers, Models, and Losses
+----------------------------------------------------
+
+A single :class:`AmpScaler` instance should scale all losses and step all optimizers.
```

Comment:
Not *should*. 

Follow-up on your previous single optimizer example. Something like:

 "Working with multiple optimizers is just like working with a single optimizer. It's important, however, that update() only be called after...